File: JavaScript solutions/921. Minimum Add to Make Parentheses Valid.js
/**
 * @param {string} s
 * @return {number}
 */
var minAddToMakeValid = function (s) {
  let sto = [];
  let stc = [];

  for (let i = 0; i < s.length; i++) {
    if (s[i] === "(") {
      sto.push(s[i]);
    } else {
      if (sto.length !== 0) {
        sto.pop();
      } else {
        stc.push(s[i]);
      }
    }
  }

  return sto.length + stc.length;
};

File: C++ solutions/921. Minimum Add to Make Parentheses Valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        stack<char>sto,stc;
        for(int i=0;i<s.size();i++){
            if(s[i]=='('){
                sto.push(s[i]);
            }else{
                if(!sto.empty()){
                    sto.pop();
                }
                else stc.push(s[i]);
            }
        }
        return sto.size()+stc.size();
    }
};